app/helpers/templates_helper.rb

```diff
@@ -46,4 +46,19 @@ def locked_warning(template)
 
     alert(:class => 'alert-warning', :text => warning_text.html_safe)
   end
+
+  def self.template_help_text
+    {
+      "PXELinux" => N_("Deployed to the TFTP server to ensure the Host boots the correct installer using pxelinux.0 file"),
+      "PXEGrub" => N_("Deployed to the TFTP server to ensure the Host boots the correct installer using grub/bootx64.efi file"),
+      "PXEGrub2" => N_("Deployed to the TFTP server to ensure the Host boots the correct installer using grub2/grubx64.efi file"),
+      "iPXE" => N_("Used in {g,i}PXE environments in place of PXELinux"),
+      "provision" => N_("The main unattended installation file; e.g. Kickstart or Preseed"),
+      "finish" => N_("A post-install script used to take custom actions after the main provisioning is complete"),
```

Comment:
we should probably mention that for kickstart system, this is only used for image based provisioning when the image does is not configured to use user-data while for preseed systems, this runs also during network based provisioning

Reply:
Perhaps:

A post-install script for cloud instances without "user data" flag set or preseed PXE systems. Connection is made via SSH, credentials or key must exist and inventory IP address must match.